adapters/get_all_characters_controller_test.go

```diff
@@ -0,0 +1,94 @@
+package adapters
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/SergioSanchezWizeline/academy-go-q12021/domain/model"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/interfaces"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/mocks"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/usecase"
+	"github.com/gorilla/mux"
+)
+
+func TestNewGetAllCharactersController(t *testing.T) {
+	type args struct {
+		usecase usecase.GetAllCharactersUserCase
+	}
+	tests := []struct {
+		name string
+		args args
+		want interfaces.Controller
+	}{
+		{
+			"Create controller success",
+			args{&mocks.GetAllCharactersUserCase{}},
+			&getAllCharactersController{&mocks.GetAllCharactersUserCase{}},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewGetAllCharactersController(tt.args.usecase); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewGetAllCharactersController() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getAllCharactersController_ServeHTTP(t *testing.T) {
+	tests := []struct {
+		name     string
+		expected string
+		wantErr  bool
+	}{
+		{
+			"Get all values",
+			`[{"id":1,"name":"Name"}]`,
+			false,
+		},
+		{
+			"should return error on ",
+			``,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			usecase := &mocks.GetAllCharactersUserCase{}
+			if tt.wantErr {
+				usecase.On("Execute").Return(nil, errors.New("error"))
```

Comment:
I don't have experience using this testing framework, so will this still work, given that the function `Execute` was renamed to `ServeHTTP`?

Reply:
Yes, the ServerHTTP is for the controller, the use case still use Execute